src/full_node/address_manager_store.py

```diff
@@ -0,0 +1,231 @@
+        await cursor.close()
+        return [
+            (node_id, ExtendedPeerInfo.from_string(info_str))
+            for node_id, info_str in nodes_id
+        ]
+
+    async def get_new_table(self):
+        cursor = await self.db.execute("SELECT node_id, bucket from peer_new_table")
+        entries = await cursor.fetchall()
+        await cursor.close()
+        return [(node_id, bucket) for node_id, bucket in entries]
+
+    async def set_metadata(self, metadata):
+        for key, value in metadata:
+            cursor = await self.db.execute(
+                "INSERT OR REPLACE INTO peer_metadata VALUES(?, ?)",
+                (key, value),
+            )
+            await cursor.close()
+            await self.db.commit()
+
+    async def set_nodes(self, node_list):
+        for node_id, peer_info in node_list:
+            cursor = await self.db.execute(
+                "INSERT OR REPLACE INTO peer_nodes VALUES(?, ?)",
+                (node_id, peer_info.to_string())
+            )
+            await cursor.close()
+            await self.db.commit()
+
+    async def set_new_table(self, entries):
+        for node_id, bucket in entries:
+            cursor = await self.db.execute(
+                "INSERT OR REPLACE INTO peer_new_table VALUES(?, ?)",
+                (node_id, bucket),
+            )
+            await cursor.close()
+            await self.db.commit()
+
+    async def serialize(self, address_manager: AddressManager):
+        metadata = []
+        nodes = []
+        new_table_entries = []
+
+        metadata.append(("key", str(address_manager.key)))
+        metadata.append(("new_count", str(address_manager.new_count)))
+        metadata.append(("tried_count", str(address_manager.tried_count)))
+
+        unique_ids = {}
+        count_ids = 0
+
+        for node_id, info in address_manager.map_info.items():
+            unique_ids[node_id] = count_ids
+            if info.ref_count > 0:
+                assert count_ids != address_manager.new_count
+                nodes.append((count_ids, info))
+                count_ids += 1
+
+        tried_ids = 0
+        for node_id, info in address_manager.map_info.items():
+            if info.is_tried:
+                assert info is not None
+                assert tried_ids != address_manager.tried_count
+                nodes.append((count_ids, info))
+                count_ids += 1
+                tried_ids += 1
+
+        for bucket in range(NEW_BUCKET_COUNT):
+            for i in range(BUCKET_SIZE):
+                if address_manager.new_matrix[bucket][i] != -1:
+                    index = unique_ids[address_manager.new_matrix[bucket][i]]
+                    new_table_entries.append((index, bucket))
+
+        await self.clear()
+        await self.set_metadata(metadata)
+        await self.set_nodes(nodes)
+        await self.set_new_table(new_table_entries)
+
+    async def unserialize(self) -> AddressManager:
```

Comment:
The more common form of the word that means the opposite of `serialize` is `deserialize`.